dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/exception/AmazonS3RetryableResponseErrorHandler.java

```diff
@@ -35,29 +36,27 @@
 
   @Override
   public void handleError(ClientHttpResponse response) throws IOException {
-    AmazonS3Exception e = new AmazonS3Exception("s3 exception", IOUtils.toString(response.getBody()));
-    switch (response.getStatusCode()) {
+    HttpStatus status = response.getStatusCode();
+    AmazonS3Exception e = new AmazonS3Exception(IOUtils.toString(response.getBody()));
+    switch (status) {
     case BAD_REQUEST:
       if (e.getErrorCode() == null || e.getErrorCode().equals("RequestTimeout")) {
-        throw new RetryableException(new IOException("Amazon S3 error: "
-            + IOUtils.toString(response.getBody())));
+        throw new RetryableException(new IOException(IOUtils.toString(response.getBody())));
       }
     case NOT_FOUND:
-      throw new NotRetryableException(new IOException("Amazon S3 error: "
-          + IOUtils.toString(response.getBody())));
+      throw new NotRetryableException(new IOException(IOUtils.toString(response.getBody())));
     case INTERNAL_SERVER_ERROR:
       log.warn("Server error. Stop processing: {}", response.getStatusText());
-      throw new NotResumableException(new IOException("Amazon S3 error: "
-          + IOUtils.toString(response.getBody())));
+      throw new NotResumableException(new IOException(IOUtils.toString(response.getBody())));
     case FORBIDDEN:
       log.warn("FORBIDDEN response code received");
-      throw new NotRetryableException(new IOException(
-          "Amazon S3 error: Access refused by object store. Confirm request host is on permitted list"
-              + IOUtils.toString(response.getBody())));
+      throw new NotRetryableException(
+          new IOException(
```

Comment:
Maybe we are not understanding each other, I see a lot of this repeating:

```java
throw new NotResumableException(new IOException(IOUtils.toString(response.getBody())));
```

We can significantly cut down the noise if we add a helper(s):

```java
throwNotResumableIO(response, "Optional message");
```

Same for retryable. In general, we should try to avoid repeating code like this as when it changes, it changes together and the opportunity to miss something is high. Also, it makes it harder to groke.